### PR TITLE
 #724: Update rewrite rules docs

### PR DESCRIPTION
**Pull request by ysusuk**

### CODE REVIEW

File: readme/Configuration.scalatex [hunk 1]
```diff
@@ -423,6 +423,27 @@
       @b{NEW}. Semantically preserving rewrites. To enable a
       rewrite rule, add it to the config like this @b{rewrite.rules = [SortImports]}.
 
+    @sect(Rewrite.rewrite2name(AvoidInfix))
+      @demoStyle(rewriteInfix)
+        future recover {
+          case e: ArithmeticException ⇒ 0
+        } map (_.toString)
+
```
Comment (olafurpg) [inline readme/Configuration.scalatex]: Can we also document the include/exclude filters?

File: readme/Configuration.scalatex [hunk 2]
```diff
@@ -423,6 +423,27 @@
       @b{NEW}. Semantically preserving rewrites. To enable a
```
Comment (olafurpg) [inline readme/Configuration.scalatex]: While you're at it, can you remove "NEW"?

File: readme/Configuration.scalatex [hunk 3]
```diff
@@ -423,6 +423,27 @@
...
+        import a.{
+            b,
+            c
+          }, h.{
+            k, l
+          }
+          import d.e.{f, g}
+          import a.{
+              foo => bar,
+              zzzz => _
```
Comment (olafurpg) [inline readme/Configuration.scalatex]: What happens in this case?

```scala
import a.{b => c, d => _, _}
// this is OK
import a.{b => c, d => _, _}
// this is bug
import a.{b => c}
import a.{d => _}
import a. _
```

If it's a bug then we should open a ticket.

Comment (ysusuk) [inline readme/Configuration.scalatex]: https://github.com/olafurpg/scalafmt/issues/726

Comment (olafurpg) [inline readme/Configuration.scalatex]: Thanks! I bumped into this recently while implementing "organize imports" in scalafix. Didn't know renames and unimports needs to be bundled together with the wildcard.

**Pull request merged**